%YAML 1.2
# SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
# Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
# SPDX-License-Identifier: Apache-2.0
---
name: Image Format Converter Substitution Rule
operation: substitution
body:
  groups:
    - entities:
        - components:
          - type: nvidia::cvcore::tensor_ops::ConvertColorFormat
            parameters:
              output_type: "BGR_U8"
              receiver: data_receiver
              transmitter: data_transmitter
              pool: pool
              input_adapter: resource/adapter
              output_adapter: resource/adapter
              output_name: "image"
            overriding_parameters: []
    - entities:
        - components:
          - type: nvidia::cvcore::tensor_ops::StreamConvertColorFormat
            parameters:
              output_type: "NV24"
              receiver: data_receiver
              transmitter: data_transmitter
              pool: pool
              input_adapter: resource/adapter
              output_adapter: resource/adapter
              output_name: "image"
              stream: resource/stream
            overriding_parameters: [output_type]
